gpflow/utilities/utilities.py

```diff
@@ -420,6 +420,11 @@ def traverse_module(
             new_acc = (f"{path}['{term_idx}']", new_state)
             new_state = traverse_module(subterm, new_acc, update_cb, target_types)
     elif isinstance(m, tf.Module):
+        items = vars(m)
+        if isinstance(m, tfp.bijectors.Bijector) and "_parameters" in items:
+            assert items["_parameters"]["self"] is m
+            del items["_parameters"]["self"]  # avoid infinite recursion
+            del items["_parameters"]  # fix other bug
```

Comment:
This is needed specifically to make the `test_freeze` happy. We could potentially move this into the `freeze()` below.

Reply:
More specifically, the set_attr_by_path doesn't work with the `_parameters` attribute. Of course it's redundant to delete just the `"self"` entry and the whole dictionary - my intent was to highlight that there are two slightly separate issues. We can resolve both in one go by deleting `_parameters` entirely, or we could only delete its `"self"` entry and separately fix the other issue.

Reply:
since `vars` returns the __dict__ of an object and as this isnt used in the line below - we can see its modifying the object. Lets take a copy of the vars object and delete from that instead!